src/omv/ports/stm32/sensor.c

```diff
@@ -976,6 +976,7 @@ int sensor_snapshot(sensor_t *sensor, image_t *image, uint32_t flags) {
         case PIXFORMAT_BAYER:
             MAIN_FB()->pixfmt = PIXFORMAT_BAYER;
             MAIN_FB()->subfmt_id = sensor->hw_flags.bayer;
+            MAIN_FB()->pixfmt = imlib_bayer_shift(MAIN_FB()->pixfmt, MAIN_FB()->x, MAIN_FB()->y, sensor->transpose);
```

Comment:
```suggestion
            MAIN_FB()->pixfmt = imlib_bayer_shift(PIXFORMAT_BAYER, MAIN_FB()->x, MAIN_FB()->y, sensor->transpose);
```

Note you're setting `MAIN_FB()->pixfmt = PIXFORMAT_BAYER;` twice, line 977 should be removed, the same goes for the other file. Otherwise this looks good to merge.

Reply:
@iabdalkader - It's not the same. pixfmt includes the subfmt_id as a field under it. Your suggestion would zero the subfmt_id.

Reply:
You're right, I forgot how that worked.